src/jdk.jpackage/share/classes/jdk/jpackage/internal/DottedVersion.java

```diff
@@ -148,8 +148,7 @@ public int compareTo(String o) {
         if (components.isEmpty()) {
             components.add(BigInteger.ZERO);
         }
-        return components.stream()
-                .collect(Collectors.toList()).toArray(BigInteger[]::new);
+        return components.stream().toList().toArray(BigInteger[]::new);
```

Comment:
I guess this can be simplified down to `components.stream().toArray(BigInteger[]::new);`

Reply:
Good catch!